base/pkg/entry.jl

```diff
@@ -155,31 +156,55 @@ end
 
 status(io::IO, pkg::AbstractString) = status(io, pkgname = pkg)
 
-function status(io::IO, pkg::AbstractString, ver::VersionNumber, fix::Bool)
+function status(io::IO, pkg::AbstractString, ver::VersionNumber, fix::Bool, avail::Dict, instd::Dict)
     @printf io " - %-29s " pkg
-    fix || return println(io,ver)
-    @printf io "%-19s" ver
-    if ispath(pkg,".git")
-        prepo = GitRepo(pkg)
-        try
-            with(LibGit2.head(prepo)) do phead
-                if LibGit2.isattached(prepo)
-                    print(io, LibGit2.shortname(phead))
+    if !fix
+        maxver = maximum(keys(avail[pkg]))
```

Comment:
the max available package version for any julia version may have completely different requirements than the max available package version that supports the in-use julia version

Reply:
I see it as a good thing because it'll inform users about changes in the package.

Reply:
If there are updates of the package that only support versions of Julia strictly newer than what the user is running, this code isn't looking at the correct dependencies to determine the information it's trying to show.

Reply:
Main point of this PR is to provide more information to the user. I do not think it is a good idea to constrain version information to the specific version of julia. After all, it might force the user to update julia.

Reply:
Dependencies of package versions that the user is not allowed to use will never be the right information to show the user here - you want to show them which dependencies of the most recent package versions *they could possibly use* might be held back.